Makefile

```diff
@@ -210,6 +210,18 @@ build ${PROTOC_GEN_SWIFT}:
 	@rm Tests/LinuxMain.swift.new
 	${SWIFT} build
 
+# This will get run by any other rule that tries to use the plugin.  This hacks
+# in a check during the library build to ensure that the protoc on the local
+# system is one we expect. This was inspired by the findings that lead to
+#   https://github.com/apple/swift-protobuf/issues/111
+# This likely will need follow up to eventually do a pattern match to ensure it
+# is 3.1 or higher.
+${PROTOC_GEN_SWIFT}: build
+	@if [ "$(shell ${PROTOC} --version)" != "libprotoc 3.1.0" ]; then \
```

Comment:
Changing the != check to a grep for a few different version number patterns might not be that bad? I'll leave it up to you whether you want to try it now or file an issue and clean it up later.